CONTRIBUTING.md

```diff
@@ -72,6 +72,20 @@ If you are modifying one or more of the NPM packages in the `/packages` director
 GitHub provides additional document on [forking a repository](https://help.github.com/articles/fork-a-repo/) and
 [creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
 
+## Keeping code generation in sync with aws-sdk-js-v3
+
+Any changes made to the `smithy-typescript-codegen` package need to be compatible with aws-sdk-js-v3. Maintainers and reviewers
+MUST ensure that code generation is kept in sync between the two repositories by creating an equivalent PR in aws-sdk-js-v3.
+
+Procedure to keep code generation in sync while making changes to `smithy-typescript-codegen`:
+1. Fork and clone [aws/aws-sdk-js-v3][aws-sdk-js-v3].
+2. Run `yarn` to install dependencies.
+3. Run `generate-clients` with `HEAD` commit as follows:
+    ```sh
+    yarn generate-clients --commit HEAD
+    ```
+4. If the clients are updated, post a pull request on aws-sdk-js-v3. If the clients are not updated, no further action is needed.
+5. When PR on smithy-typescript is merged, rebase the PR on aws-sdk-js-v3 and add merged commit as default.
```

Comment:
If the codegen change affects new clients or client updates published after the time of running the synchronization codegen, it should be rerun.

There is a practical time constraint to doing this, since the ideal time is after a release, but including that information here may be too specific?

Reply:
> There is a practical time constraint to doing this, since the ideal time is after a release, but including that information here may be too specific?

Yup. The codegen PR can be created any time, but it needs to be rebased and merged after a release in the current setup of aws-sdk-js-v3.